packages/cli-repl/src/cli-repl.ts

```diff
@@ -147,7 +148,36 @@ class CliRepl {
       let result;
 
       try {
-        result = await this.shellEvaluator.customEval(originalEval, input, context, filename);
+        let sigintListener: () => void;
+        let previousSigintListeners: any[];
+        try {
+          result = await new Promise((resolve, reject) => {
+            // Handle SIGINT (Ctrl+C) that occurs while we are stuck in `await`
+            // by racing a listener for 'SIGINT' against the evalResult Promise.
+            // We remove all 'SIGINT' listeners and install our own.
+            sigintListener = (): void => {
+              // Reject with an exception similar to one thrown by Node.js
+              // itself if the `customEval` itself is interrupted.
+              reject(new Error('Asynchronous execution was interrupted by `SIGINT`'));
+            };
+            previousSigintListeners = this.repl.rawListeners('SIGINT');
```

Comment:
I _did not_ know you can just hook into the listeners o/

Reply:
Fwiw, this code is partially inspired by the way that Node.js does this internally for the REPL option :)

https://github.com/nodejs/node/blob/02e925553d079c1e70b48e5255503d8850eb1fc5/lib/vm.js#L264-L280